rust/Earthfile

```diff
@@ -1,24 +1,45 @@
 VERSION --global-cache 0.7
 
+# INIT stores the configuration required for the other UDCs in the filesystem, and installs required dependencies.
+#   - cache_id: Overrides default ID of the global $CARGO_HOME cache. Its value is exported to the build environment under the entry: $CARGO_HOME_CACHE_ID
+#   - keep_fingerprints (false): Instructs the following +CARGO calls to don't remove the Cargo fingerprints of the source packages. Use only when source packages have been COPYed with --keep-ts option.
+#   - sweep_days (4): +CARGO calls use cargo-sweep to clean build artifacts that haven't been accessed for this number of days.
+INIT:
+    COMMAND
+    DO +INSTALL_CARGO_SWEEP
+    RUN mkdir -p /earthly/cfg
+
+    # cache_id
+    ARG EARTHLY_TARGET_PROJECT_NO_TAG
+    ARG OS_RELEASE=$(md5sum /etc/os-release | cut -d ' ' -f 1)
+    ARG cache_id="${EARTHLY_TARGET_PROJECT_NO_TAG}#${OS_RELEASE}#earthly-cargo-cache"
+    RUN echo "$cache_id">/earthly/cfg/cache_id
+    ENV CARGO_HOME_CACHE_ID=$cache_id
+
+    #keep_fingerprints
+    ARG keep_fingerprints=false
+    RUN echo "$keep_fingerprints">/earthly/cfg/keep_fingerprints
+
+    #sweep_days
+    ARG sweep_days=4
+    RUN echo "$sweep_days">/earthly/cfg/sweep_days
+
 # CARGO runs the cargo command "cargo $args".
-# This UDC should be thread safe. Parallel builds of targets using it should be free of race conditions.
+# This UDC is thread safe. Parallel builds of targets calling this UDC should be free of race conditions.
 # Arguments:
 #   - args: Cargo subcommand and its arguments. Required.
-#   - keep_fingerprints (false): Do not remove source packages fingerprints. Use only when source packages have been COPYed with --keep-ts option.
-#   - sweep_days (4): The UDC uses cargo-sweep to clean build artifacts that haven't been accessed for this number of days.
 #   - output: Regex to match the files within the target folder to be copied from the cache to the caller filesystem (image layers).
 #     Use this argument when you want to SAVE an ARTIFACT from the target folder (mounted cache), always trying to minimize the total size of the copied fileset.
 #     For example --output="release/[^\./]+" would keep all the files in /target/release that don't have any extension.
 CARGO:
     COMMAND
     ARG --required args
-    ARG keep_fingerprints=false
-    ARG sweep_days=4
+    ARG keep_fingerprints=$(cat /earthly/cfg/keep_fingerprints)
+    ARG sweep_days=$(cat /earthly/cfg/sweep_days)
```

Comment:
Possible idea - I wonder if we should detect if `rust+INIT` was called and call it implicitly here if not.

Reply:
And if not I think every `COMMAND` which depends on this having been run ought to say so in it's doc comment?

Reply:
And the doc file itself says "It is meant to be called once per build environment..." which I think maybe ought to be "It must be called..."?

Reply:
I added checks to make it clear it has to be called once. Calling implicitly could make installing dependencies multiple times (cargo sweep)